src/mongocxx/client.cpp

```diff
@@ -66,5 +66,11 @@ class database client::database(bsoncxx::stdx::string_view name) const & {
     return mongocxx::database(*this, name);
 }
 
+cursor client::list_databases() const {
+    bson_error_t error;
+    auto result = libmongoc::client_find_databases(_impl->client_t, &error);
```

Comment:
if result is nullptr, we should throw an exception::operation containing the message and code. See collection::create_index for an example.